src/devices/chihiro/JvsIo.cpp

```diff
@@ -97,203 +118,246 @@ int JvsIo::Jvs_Command_GetBoardId()
-	for (unsigned i = 0; i < header->count; i++) {
-		uint8_t value = payload[i];
+	uint8_t value = *payload++;
 
-		// Special case: 0xD0 is an exception byte that actually returns the next byte + 1
-		if (value == 0xD0) {
-			value = payload[i++] + 1;
-		}
-
-		packet.push_back(value);
+	// Special case: 0xD0 is an exception byte that actually returns the next byte + 1
+	if (value = EscapeByte) {
+		value = *payload++ + 1;
 	}
 
+	return value;
+}
+
+void JvsIo::HandlePacket(jvs_packet_header_t* header, std::vector<uint8_t>& packet)
+{
 	// Clear the response buffer
 	ResponseBuffer.clear();
 
 	ResponseBuffer.push_back(0x01);
 	
 	// It's possible for a JVS packet to contain multiple commands, so we must iterate through it
-	// -1 is to skip the checksum byte
-	for (int i = 0; i < packet.size() - 1; i++) {
+	for (size_t i = 0; i < packet.size(); i++) {
 		BroadcastPacket = packet[i] >= 0xF0; // Set a flag when broadcast packet
 
+		uint8_t* command_data = &packet[i];
 		switch (packet[i]) {
 			// Broadcast Commands
-			case 0xF0: i += Jvs_Command_Reset(); break;
-			case 0xF1: i += Jvs_Command_SetDeviceId(&packet[i]); break;
+			case 0xF0: i += Jvs_Command_F0_Reset(command_data); break;
+			case 0xF1: i += Jvs_Command_F1_SetDeviceId(command_data); break;
 			// Init Commands
-			case 0x10: i += Jvs_Command_GetBoardId(); break;
-			case 0x11: i += Jvs_Command_GetCommandFormat();	break;
-			case 0x12: i += Jvs_Command_GetJvsRevision(); break;
-			case 0x13: i += Jvs_Command_GetCommunicationVersion(); break;
-			case 0x14: i += Jvs_Command_GetCapabilities(); break;
-			case 0x20: i += Jvs_Command_ReadSwitchInputs(&packet[i]); break;
-			case 0x21: i += Jvs_Command_ReadCoinInputs(&packet[i]); break;
-			case 0x22: i += Jvs_Command_ReadAnalogInputs(&packet[i]); break;
-			case 0x32: i += Jvs_Command_GeneralPurposeOutput(&packet[i]); break;
+			case 0x10: i += Jvs_Command_10_GetBoardId(); break;
+			case 0x11: i += Jvs_Command_11_GetCommandFormat();	break;
+			case 0x12: i += Jvs_Command_12_GetJvsRevision(); break;
+			case 0x13: i += Jvs_Command_13_GetCommunicationVersion(); break;
+			case 0x14: i += Jvs_Command_14_GetCapabilities(); break;
+			case 0x20: i += Jvs_Command_20_ReadSwitchInputs(command_data); break;
+			case 0x21: i += Jvs_Command_21_ReadCoinInputs(command_data); break;
+			case 0x22: i += Jvs_Command_22_ReadAnalogInputs(command_data); break;
+			case 0x32: i += Jvs_Command_32_GeneralPurposeOutput(command_data); break;
 			default:
 				printf("JvsIo::HandlePacket: Unhandled Command %02X\n", packet[i]);
 				break;
-
 		}
 	}
 }
 
-size_t JvsIo::SendPacket(jvs_packet_header_t* packet)
+size_t JvsIo::SendPacket(uint8_t* buffer)
 {
-	// Total Size = Packet Count + Sync, Mode, Checksum
-	size_t totalPacketSize = packet->count + 3;
-	uint8_t* payload = (uint8_t*)packet + 3;
+	// Remember where the buffer started (so we can calculate the number of bytes we've handled)
+	uint8_t* buffer_start = buffer;
 
-#ifdef DEBUG_JVS_PACKETS

```

Comment:
Why was this debug function removed?

Reply:
As it looked unused and was not easily refactored

Reply:
I'll restore it